cypress/integration/user/privacy-terms.js

```diff
@@ -0,0 +1,33 @@
+/* global cy */
+
+describe('Privacy terms', () => {
+  it('should not redirect away from email sign up page on login', () => {
+    // Flag used to identify if the `/update-privacy-terms` have been called
+    let privacyTermsUpdated = false;
+    cy.intercept('PUT', '/update-privacy-terms', () => {
+      privacyTermsUpdated = true;
+    }).as('updatePrivacyTerms');
+
+    // Seed dev user with `acceptedPrivacyTerms` unset
+    cy.exec('npm run seed -- --unset-privacy-terms');
+    // Go to the homepage and log in manually so we can assert the following:
+    // 1. Redirection to /email-sign-up works properly
+    // 2. The /update-privacy-terms has not been requested
+    cy.visit('/');
+    cy.get('[data-test-label="landing-small-cta"]').click();
+    cy.location('pathname').should('contain', '/email-sign-up');
+    cy.wrap(privacyTermsUpdated).should('eq', false);
+    // Assert email sign up elements and make sure we don't get redirected somewhere else
+    cy.title().should('contain', 'Email Sign Up');
+    cy.contains('Email Sign Up').should('exist');
+    cy.contains('Please slow down and read this.').should('exist');
```

Comment:
I think `cy.contains(...).should('exist')` is redundant, since if `cy.contains` succeeds the returned DOM element must exist.

Also, if the `title` has the text "Email Sign Up", then line 22 will always succeed.  How about adding a `data-cy` to that so we can get it directly?

Reply:
Thanks for reviewing @ojeytonwilliams. Removed `should exist` assertions for all `cy.contains`. Added a `data-cy` to the containing `Grid` and targeted it for assertion.